### ISSUE TITLE
CRA 4 build failure with the new map component 

### ISSUE THREAD

**Issue by jacoblogan** [aws-amplify/amplify-ui#1656]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

CRA 4

### Please describe your bug.

Fresh application fails to start with a compile error when using CRA 4 showing the compile error 

```
./node_modules/react-map-gl/dist/esm/components/use-control.js 12:46
Module parse failed: Unexpected token (12:46)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
| 
|     if (!map.hasControl(ctrl)) {
>       map.addControl(ctrl, (opts || onRemove)?.position);
|     }
|
```

This issue is resolved when using the current version of CRA 5.

### What's the expected behaviour?

No compile errors on run.

### Help us reproduce the bug!

using CRA 4 create a react application
`npm install aws-amplify @aws-amplify/ui-react`
import any component from the ui-react package into your application
`npm run start` to see the compile error

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

package dependencies
```
"dependencies": {
    "@aws-amplify/ui-react": "^2.15.1",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^12.1.4",
    "@testing-library/user-event": "^13.5.0",
    "aws-amplify": "^4.3.19",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-scripts": "4.0.3",
    "web-vitals": "^2.1.4"
  },
```

_No response_

**Comment by roxannelessard** [comment]
@jacoblogan We have the same problem, we spend the whole day trying to understand what was breaking our app. Now we can't release anything. Supposed to be easy-to-use and since day 1 every time we want to make a change or customize 1 small thing it takes us ages and we can't find any clear documentation.

**Comment by Milan-Shah** [comment]
We apologize you are having this issue, we are currently implementing a fix and will update here shortly!

**Comment by reesscot** [comment]
Upgrading Create React App to version 5 can mitigate this issue. We are still investigating how to fix it in CRA version 4.

*How to upgrade to version 5:*
```
npm install --save react-scripts@5
```

**Comment by slaymance** [comment]
This has been an issue with previous versions of `create-react-app` lacking support for optional chaining (see [#8445](https://github.com/facebook/create-react-app/issues/8445).

The recommended fix is to upgrade your Create React App to version 5:
```
npm install --save react-scripts@5
```

If you are unable to upgrade your `react-scripts` version, try making your browser target list less aggressive by doing the following:
1. Remove the `.cache` directory from your project's `node_modules`:
```
rm -rf node_modules/.cache
```
2. Change the `browserslist` block in your `package.json` from:
```json
"browserslist": {
  "production": [
    ">0.2%",
    "not dead",
    "not op_mini all"
  ],
  "development": [
    "last 1 chrome version",
    "last 1 firefox version",
    "last 1 safari version"
  ]
}
```
to the following:
```json
"browserslist": [
  ">0.2%",
  "not dead",
  "not op_mini all"
]
```
3. Try rebuilding and starting your project with:
```
npm run build && npm start
```

This fix was suggested in a similar [CRA issue](https://github.com/facebook/create-react-app/issues/9468#issuecomment-694191642).

**Comment by Milan-Shah** [comment]
@roxannelessard and @raffpaquin, @slaymance has provided a workaround while we are waiting for an upstream dependency to fix the issue ((see https://github.com/facebook/create-react-app/issues/8445). Please let us know if you are still having any problems. Feel free to reopen the ticket if so.

**Comment by slaymance** [comment]
This issue has been resolved in [@aws-amplify/ui-react@2.17.1](https://github.com/aws-amplify/amplify-ui/releases/tag/%40aws-amplify%2Fui-react%402.17.1) and the recommended fix now is to upgrade your dependency.
